.github/workflows/integration_test.yaml

```diff
@@ -0,0 +1,49 @@
+name: Integration tests
+
+on:
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  integration-tests-arm:
+    name: Integration test (ARM64)
+    runs-on: [self-hosted, ARM64, jammy, stg-private-endpoint]
+    strategy:
+      matrix:
+        image: [jammy, noble]
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
Can you pin this? Also, checkout 4 is available iirc

Reply:
May I ask what you mean by pinning the actions? I don't think you can pin the action versions.
As for checkout version 4, I think a follow-up PR with dependabot should work :)

Reply:
Yes, you can pin to any tag https://github.com/actions/checkout/releases/tag/v4.1.7, and we can wait for dependabot to upgrade

Reply:
pinned! thank you!